### ISSUE TITLE
Migrating to .NET Core 3.1 

### ISSUE THREAD

**Issue by HeinPauwelyn** [code4me/4me-sdk-csharp#41]
# Problem situation 

I’m creating an application with the 4me API and for that I’ll use the codebase from this GitHub repo. I’ve also created a PAT for bearer authentication. But when I call the persons of at 4me using code below.

```c#
FourMeContext fourMeContext = new FourMeContext();
Sdk4meClient fourMeClient = fourMeContext.GetFourMeClient();

IEnumerable<Person> person = fourMeClient.People.Get();
```

I’ve next error:

> **`System.Private.CoreLib`:** Exception while executing function: `<function name>`. 
>
> **`Sdk4me`:** Unexpected character encountered while parsing value: `<`. Path `''`, line 0, position 0. 
>
> **`System.Net.Requests`:** The remote server returned an error: (400) Bad Request.

Code inside the `FourMeContext`:

```C#
public class FourMeContext
{
    public Sdk4meClient GetFourMeClient(CancellationToken cancellation = default)
    {
        KeyProvider keyProvider = new KeyProvider();
        string key = keyProvider.GetSecret("4me-debug", cancellation);

        EnvironmentType environmentType = EnvironmentType.Quality;
        AuthenticationToken token = new AuthenticationToken(key, AuthenticationType.BearerAuthentication);

        return new Sdk4meClient(token, "savaco-internal-it", environmentType);
    }
}
```

The key returned from the KeyProvider is correct. When I use Fiddler to capture the request I see this:

![image](https://user-images.githubusercontent.com/16222780/114163644-b63c1880-992a-11eb-8a81-ac1c5f217de4.png)

Using Postman everything I’ve got a 200 response.

![image](https://user-images.githubusercontent.com/16222780/114163674-c3590780-992a-11eb-952b-f0cfb9b8d247.png)

# Possible fix

After searching for a while, I’ve found the problem. It comes because the target Framework of the library is .NET Framework 4.5, but we’re using .NET Core 3.1. The question is now could it be migrated to .NET Core 3.1 or newer (.NET 5 or later .NET 6)? When creating a new solution targeting .NET Framework 4.8 it works.

Hoping for a quick response.

**Comment by klaasvandeweerdt** [comment]
@HeinPauwelyn, I never used it on .NET core. I will play around on 3.1 and 5 and update the existing code. Any input is welcome.

**Comment by klaasvandeweerdt** [comment]
@HeinPauwelyn, found the issue. The way that the web request is generated is different in .NET Framework and .NET Core.
.NET Core sets the content-length attribute automatically 0 as soon as a content-type is set. This does not happen in .NET Framework,
The `BuildWebRequest` method currently contains the following code `request.ContentType = "application/json";`
Adapting the method and inserting the `request.ContentType = "application/json";` in the `Insert` and `Update` method solves the problem. All methods can be found the the `BaseHandler<T, X>` class.

I haven't tested it properly but first test show that this solved it and the code works on Framework 4.8 and Core 3.1.
Once properly tested I will publish the updated code.

**Comment by klaasvandeweerdt** [comment]
Tests are completed and an update has been pushed. The solution is still using .NET Framework 4.5, but the code base can be used in .NET Core 3.1 or higher.
